lib/redis/connection/ruby.rb

```diff
@@ -3,12 +3,22 @@
 require "redis/errors"
 require "socket"
 
+begin
+  require "openssl"
+rescue LoadError
+  # Not all systems have OpenSSL support
+end
+
 class Redis
   module Connection
     module SocketMixin
 
       CRLF = "\r\n".freeze
 
+      # Exceptions raised during non-blocking I/O ops that require retrying the op
+      NBIO_EXCEPTIONS = [Errno::EWOULDBLOCK, Errno::EAGAIN]
+      NBIO_EXCEPTIONS << IO::WaitReadable if RUBY_VERSION >= "1.9.3"
```

Comment:
@tarcieri Thoughts on using nio4r here instead of re-implementing the `IO.select` logic around `IO::WaitReadable` and `IO::WaitWritable`>

We are seeing `OpenSSL::SSL::SSLErrorWaitWritable` errors when writing large keys. cc @mdayaram